### PR TITLE
CI

### PR DESCRIPTION
**Pull request by sgasioro**
First round of CI: reproducibility test for output of forward model + gradients for backward pass, with readout noise turned off. Removed always failing docs test. Changes beyond the tests:
- Removed LFS usage -- can reintroduce, but caused some weird things with the test environment
- Introduced readout_noise toggle -- need for reproducibility
- Introduced seed for track selection -- need for reproducibility
- Removed dz requirement on track segments -- need for lightweight tests
- Removed time tick padding (`len(sim.time_ticks)*50`-> `*3` -- maybe should revert @YifanC ?)
- Fix dtype inheritance -- doubles => no floating point discrepancy

### CODE REVIEW

File: optimize/dataio.py [hunk 1]
```diff
@@ -55,14 +52,15 @@ def __init__(self, filename, ntrack, swap_xz=True):
...
+            random.seed(seed)
             list_rand = random.sample(range(len(index)), ntrack)
             for i_rand in list_rand:
                 fit_index.append(index[i_rand])
                 fit_tracks.append(all_tracks[i_rand])
             self.tracks = torch.nn.utils.rnn.pad_sequence(fit_tracks, batch_first=True, padding_value = -99) 
         
         #self.tracks = torch.nn.utils.rnn.pad_sequence(all_tracks, batch_first=True, padding_value = -99) 
-        print("trainning set [ev, trk]: ", fit_index)
+        print("training set [ev, trk]: ", fit_index)
```
Comment (ynashed) [inline optimize/dataio.py:63]: Spelling police.

**Review by ynashed — APPROVED** [review]

File: optimize/dataio.py (hunk 1 above)
Comment (sgasioro) [reply optimize/dataio.py:63]: Lol, couldn't help myself

File: optimize/utils.py [hunk 2]
```diff
@@ -52,7 +52,7 @@ def all_sim(sim, selected_tracks, fields, event_id_map, unique_eventIDs, return_
                                           fields=fields)
 
     unique_pix_torch = torch.empty((0, 2), device=neighboring_pixels_torch.device)
-    pixels_signals_torch = torch.zeros((len(unique_pix_torch), len(sim.time_ticks)*50),
+    pixels_signals_torch = torch.zeros((len(unique_pix_torch), len(sim.time_ticks)*3),
```
[the remark this answers is not in the document]
Comment (sgasioro) [reply optimize/utils.py:55]: The `50` there is an arbitrary padding to allow us to distribute different events in time (1 event has `len(sim.time_ticks)` ticks, with the `*3` as an over/underflow padding). I think this is a change that we eventually want to make, but @YifanC is much more qualified to speak on this -- might be that I should save this for a future PR.

[the remark this answers is not in the document]
Comment (YifanC) [reply optimize/utils.py:55]: @sgasioro Maybe keep `*50` here and change it to `*3` once we move to event loss calculation?
